networkx/algorithms/isomorphism/tests/test_tree_isomorphism.py

```diff
@@ -0,0 +1,285 @@
+import networkx as nx
+import random
+import time
+from networkx.classes.function import is_directed
+
+from networkx.algorithms.isomorphism.tree_isomorphism import (
+    rooted_tree_isomorphism,
+    tree_isomorphism,
+)
+
+
+# have this work for graph
+# given two trees (either the directed or undirected)
+# transform t2 according to the isomorphism
+# and confirm it is identical to t1
+# randomize the order of the edges when constructing
+def check_isomorphism(t1, t2, isomorphism):
+
+    # get the name of t1, given the name in t2
+    mapping = {v2: v1 for (v1, v2) in isomorphism}
+
+    # these should be the same
+    d1 = is_directed(t1)
+    d2 = is_directed(t2)
+    assert d1 == d2
+
+    edges_1 = []
+    for (u, v) in t1.edges():
+        if d1:
+            edges_1.append((u, v))
+        else:
+            # if not directed, then need to
+            # put the edge in a consistent direction
+            if u < v:
+                edges_1.append((u, v))
+            else:
+                edges_1.append((v, u))
+
+    edges_2 = []
+    for (u, v) in t2.edges():
+        # translate to names for t1
+        u = mapping[u]
+        v = mapping[v]
+        if d2:
+            edges_2.append((u, v))
+        else:
+            if u < v:
+                edges_2.append((u, v))
```

Comment:
The ordering of nodes may be different for the mapped version of the network.
It’s not clear tome that this will work correctly.  Could this be tested by checking an isomorphism that e.g. reverses the order of nodes for some edge?

Reply:
I think that's the correct way to test for isomorphism.  Two graphs are isomorphic if and only if they have same set of edges after relabeling one graph the isomorphism.   https://mathworld.wolfram.com/IsomorphicGraphs.html

In the case of a directed graph, the edge order clearly matters. But on an undirected graph, the edge (u,v) is the same as the edge (v,u).  That's why I put the edge in particualr form of u < v, before testing that the set of edges is the same. 

In the routine positive_single_tree, used in testing, I randomly shuffle the nodes before constructing a test graph. You're right that I should probably shuffle the edge order as well, to ensure it is correct both ways.  I've added a random_swap routine to do that. 

There isn't really a good proof of correctness for this algorithm.  The Aho, Hopcroft, and Ullman reference doesn't even really state the algorithm completely.  And once it was in that book, nobody seems to have published about it.  So the testing seems very important.  I tested the positive case for trees up to 14 nodes, and the negative case up to 11.

Reply:
I just mean that this ```check_isomorphism``` function uses the order ```u<v``` for t1 and ``` mapping of u < mapping of v``` for t2.
Maybe use ```new_u = mapping[u]``` (and v) and then check ```u<v``` as you do... or save the orig u and v before mapping and check ```orig_u < orig_v```.  It doesn't matter to me what way you arrange the variables-- but the node order for edges in the two edge collections should be the same or the function will report (1,2) as different from (2,1) in an undirected graph.  I feel like I'm not being clear, so ask if something doesn't make sense.

Reply:
Ahhh....     My bad!   You **DO** want to check the order of the mapped nodes... NOT the order before mapping.
Sorry about that.